SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -167,6 +167,76 @@ void TasmanianTridiagonalSolver::decompose(int n, double d[], double e[], double
     }
 }
 
+void TasmanianFourierTransform::dft(const int rank, const int n[], const double data[], std::complex<double> out[]) {
+    // Fourier transform code by ZBM
+    // Divides Fourier coeffs by normalization constant (n[0] * n[1] * ... * n[rank-1])
+
+    int num_data = 1;
+    for(int r=0; r<rank; r++) {
+        num_data *= n[r];
+    }
+
+    std::complex<double> i(0,1);
+
+
+    int k_idx[rank];
+    std::fill(k_idx, k_idx+rank, 0);
+    int k_restart_dim = 1;
+    std::complex<double>* exp_k = new std::complex<double>[rank];
+    std::fill(exp_k, exp_k+rank, 1);
+
+    for(int k=0; k<num_data; k++) {
```

Comment:
* this is the $O(n^2)$ algorithm, right? If so, we need to think about this.

Reply:
Yes, the algorithm I wrote is $O(n^2)$ (one-time at the front end, but could still be bad for grids with large depth).